src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -198,12 +201,14 @@ public bool Join(int millisecondsTimeout)
             return JoinInternal(this, millisecondsTimeout);
         }
 
+        [MemberNotNull(nameof(_waitInfo))]
         private void Initialize()
         {
             InitInternal(this);
 
-            // TODO: This can go away once the mono/mono mirror is disabled
-            stack_size = _startHelper!._maxStackSize;
```

Comment:
I think you also need to delete the `stack_size` field and pass the `stack_size` as argument instead.

Reply:
I pinged Aleksey about this TODO since I don't have context for the recent Thread changes and don't want to spent half an hour looking through it all if possible. I'll update tomorrow after I talk with him.

Reply:
This is currently used by the runtime.

Reply:
> I pinged Aleksey about this TODO since I don't have context for the recent Thread changes and don't want to spent half an hour looking through it all if possible. I'll update tomorrow after I talk with him.

I don't remember what that comment means anymore except the superficial "once we stop the mirror, the layout of `MonoInternalThread` can change".

But yea we so still need to pass a stack_size down into the runtime so that it's threaded down to the native thread creation functions. (otherwise it ends up initialized to 0 which I think mean "default size")

Reply:
I have added this comment one month ago as I was moving more Thread.cs to be shared. It is exactly as @lambdageek says - you need to pass a stack_size down into the native thread creating function.

Reply:
Hopefully I understood correctly how the stack size stuff is supposed to work?